.github/workflows/run-tests.yml

```diff
@@ -0,0 +1,45 @@
+# This workflow will install Python dependencies and run tests with PyTest using Python 3.8
+# For more information see: https://help.github.com/actions/language-and-framework-guides/using-python-with-github-actions
+
+name: Run tests
+
+on:
+  push:
+    branches: [main]
+  pull_request:
+    branches: [main]
+
+jobs:
+  build:
+    runs-on: ubuntu-latest
+
+    steps:
+      # Checkout repository
+      - name: Checkout code
+        uses: actions/checkout@v3
+
+      # Set Python version
+      - name: Set up Python 3.8
+        uses: actions/setup-python@v4
+        with:
+          python-version: 3.8
+
+      # Set up submodules
+      - name: Set up all submodules and project dependencies
+        run: |
+          git submodule update --init --recursive --remote
+
+      # Install python dependencies for each project
```

Comment:
`# Install project dependencies`

Unfortunately can't use `common` (naming this repository `common` was a mistake).